src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala

```diff
@@ -123,11 +173,15 @@ case class DataFiles(bag: DansBag) extends DebugEnhancedLogging {
    * @param path the relative path of the file or directory to delete
    */
   def delete(path: Path): Try[Unit] = {
-    val file = bag.data / path.toString
-    if (!file.exists) Failure(NoSuchFileInDepositException(file, path))
-    else (if (file.isDirectory) removeDir(file.walk().toStream)
-          else bag.removePayloadFile(path)
-      ).flatMap(_.save)
+    val file = uploadRoot / path.toString
+    if (!file.exists) {
+      Failure(NoSuchFileInDepositException(path))
+    }
+    else {
+      val triedBag = if (file.isDirectory) removeDir(file.walk().toStream)
```

Comment:
IMO `listRecursively` is more descriptive than `walk`, even though they do the same thing.

```suggestion
      val triedBag = if (file.isDirectory) removeDir(file.listRecursively().toStream)
```